torch/_dynamo/variables/torch.py

```diff
@@ -349,7 +351,7 @@ def call_function(
             torch.amp.autocast_mode.autocast,
             torch.cuda.amp.autocast,
             torch.cpu.amp.autocast,
-        ):
+        ) + tuple(([custom_backend_mod.amp.autocast,] if custom_backend_mod else [])):
```

Comment:
Consider adding a new function in _dynamo/utils.py - `is_amp_autocast` that does the custom_backend calculation once (perhaps use lru_cache).